test/framework/src/main/java/org/opensearch/index/shard/IndexShardTestCase.java

```diff
@@ -1211,6 +1211,7 @@ public void getSegmentFiles(
                 ReplicationCheckpoint.empty(replica.shardId),
                 replica,
                 source,
+                false,
```

Comment:
All these tests pass a value of false, we need a test where this flag is true and we request diff.different files?

Reply:
Yes, will add a new test to verify diff.different files are copied.